python/ray/llm/_internal/batch/stages/vllm_engine_stage.py

```
@@ -211,8 +234,24 @@ async def _prepare_llm_request(self, row: Dict[str, Any]) -> vLLMEngineRequest:
                         self.lora_name_to_request[lora_name] = lora_request
             lora_request = self.lora_name_to_request[lora_name]
 
+        # Prepare sampling parameters.
         if self.task_type == vLLMTaskType.GENERATE:
-            params = vllm.SamplingParams(**row.pop("sampling_params"))
+            sampling_params = row.pop("sampling_params")
+            if "guided_decoding" in sampling_params:
+                if self.vllm_use_v1:
+                    raise ValueError("Guided decoding is only supported with vLLM v0")
```

Non-blocker: I don't really know how this works, but would raising an error here break everything? Would it make model sense to set `guided_decoding` to None and log a warning?

---

I'd prefer to block everything and fail early; otherwise users may spend time on waiting the execution, and finally found that the outputs don't obey the JSON schema.